src/thunderbots/software/util/codec.cpp

```diff
@@ -0,0 +1,326 @@
+        }
+
+        // Remove the exponent (-1022) and shift the result up into the integer
+        // part (52 bits).
+        x = std::ldexp(x, 1022 + 52);
+
+        // Encode the significand by converting the result to an integer.
+        uint64_t significand = static_cast<uint64_t>(x + 0.5);
+
+        // Encode the number.
+        return pack_ses64(sign, 0x000, significand);
+    }
+    else
+    {
+        // Extract sign bit and absolutize input.
+        bool sign = !!std::signbit(x);
+        if (sign)
+        {
+            x = -x;
+        }
+
+// Extract exponent.
+#if FLT_RADIX == 2
+        int16_t exponent = static_cast<int16_t>(std::ilogb(x));
+#else
+        int16_t exponent = static_cast<int16_t>(std::floor(std::log2(x)));
+#endif
+
+        // If exponent is below minimum limit, encode a zero.
+        if (exponent < -1022)
+        {
+            return pack_ses64(sign, 0x000, 0);
+        }
+
+        // If exponent is above maximum limit, encode an infinity.
+        if (exponent > 1023)
+        {
+            return pack_ses64(sign, 0x7FF, 0);
+        }
+
+        // Remove the exponent.
+        x = std::ldexp(x, -exponent);
+
+        // The leading 1 bit of the significand is implied; remove it.
+        x -= 1.0;
+
+        // Shift the value up into the integer part.
+        x = std::ldexp(x, 52);
+
+        // Encode the significand by converting the result to an integer.
+        uint64_t significand = static_cast<uint64_t>(x + 0.5);
+
+        // Bias the exponent.
+        exponent = static_cast<int16_t>(exponent + 1023);
+
+        // Encode the number.
+        return pack_ses64(sign, static_cast<uint16_t>(exponent), significand);
+    }
+}
+
+double decode_u64_to_double(uint64_t x)
+{
+    // Extract the sign bit, biased exponent, and significand.
+    bool sign            = !!(x & UINT64_C(0x8000000000000000));
+    int16_t exponent     = (x >> 52U) & 0x7FFU;
+    uint64_t significand = x & UINT64_C(0x000FFFFFFFFFFFFF);
+
+    // Break down the possible exponents by class.
+    if (exponent == 0x7FF)
+    {
+        // Exponent 0x7FF encodes NaNs and infinities.
+        if (significand)
+        {
+            return NAN;
+        }
+        else if (sign)
+        {
+            return -INFINITY;
+        }
+        else
+        {
+            return INFINITY;
+        }
+    }
+    else if (!exponent)
+    {
+        // Exponent 0x000 encodes zeroes and subnormals.
+        // Shift the significand down into the fraction part.
+        double value = static_cast<double>(significand) / (UINT64_C(1) << 52);
+
+        // Apply the exponent (this value is 2^1022).
+        value /=
+            44942328371557897693232629769725618340449424473557664318357520289433168951375240783177119330601884005280028469967848339414697442203604155623211857659868531094441973356216371319075554900311523529863270738021251442209537670585615720368478277635206809290837627671146574559986811484619929076208839082406056034304.0;
```

Comment:
what the actual-

Reply:
really tho. Perhaps `const some_var_name = std::pow(2,1022)` _might_ be a bit better........?

Reply:
Holy...

Reply:
removed the file, this is not an issue anymore

Reply:
well that's one way to solve the problem..............